addons/promote/common/src/main/java/org/commonjava/indy/promote/validate/PromotionValidationTools.java

```diff
@@ -527,30 +535,33 @@ public Transfer getTransfer( final StoreResource resource )
     {
         final Logger logger = LoggerFactory.getLogger( this.getClass() );
         logger.trace( "Exe parallel on collection {} with closure {}", collection, closure );
-        collection.parallelStream().forEach( e -> {
+        collection.forEach( e -> ruleParallelExecutor.execute( () -> {
             logger.trace( "The paralleled exe on element {}", e );
             closure.call( e );
-        } );
+        } ) );
     }
 
     public <T> void paralleledEach( T[] array, groovy.lang.Closure closure )
     {
         final Logger logger = LoggerFactory.getLogger( this.getClass() );
         logger.trace( "Exe parallel on array {} with closure {}", array, closure );
-        Arrays.stream( array ).parallel().forEach( e -> {
-            logger.trace( "The paralleled exe on element {}", e );
-            closure.call( e );
-        } );
+        for ( T e : array )
+        {
+            ruleParallelExecutor.execute( () -> {
+                logger.trace( "The paralleled exe on element {}", e );
+                closure.call( e );
+            } );
+        }
```

Comment:
Hm, we might need to use a CountDownLatch to prevent returning too quickly from these calls. If we return before they're done, the whole validation rule might succeed when these tasks would have found an error otherwise.

Reply:
Sorry, forgot the latch thing. Added in new commit.